src/makeInfinityAdder.js

```diff
@@ -4,7 +4,19 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(number) {
+    if (number === undefined) {
+      return sum;
```

Comment:
When returning the sum, the current implementation does not reset the sum to 0. After adder() is called without arguments, it should reset the sum to 0 to ensure subsequent calls start with a clean state as per the task description.